kinesis/kinesis.go

```diff
@@ -199,11 +197,10 @@ func (outputPlugin *OutputPlugin) AddRecord(record map[interface{}]interface{},
 	newRecordSize := len(data) + len(partitionKey)
 
 	if len(outputPlugin.records) == maximumRecordsPerPut || (outputPlugin.dataLength+newRecordSize) > maximumPutRecordBatchSize {
-		err = outputPlugin.sendCurrentBatch()
+		retCode, err := outputPlugin.sendCurrentBatch()
 		if err != nil {
```

Comment:
There's a case in `processAPIResponse` where FLB_RETRY is returned but the error is nil; need to change this if block to account for that, or change that code